Mage/src/main/java/mage/abilities/SpellAbility.java

```diff
@@ -146,7 +146,7 @@ public ActivationStatus canActivate(UUID playerId, Game game) {
             if (getSpellAbilityType() == SpellAbilityType.BASE_ALTERNATE) {
                 Player player = game.getPlayer(playerId);
                 if (player != null
-                        && player.getCastSourceIdWithAlternateMana().contains(getSourceId())) {
+                        && player.getCastSourceIdWithAlternateMana().getOrDefault(getSourceId(), new HashSet<>()).contains(null)) {
```

Comment:
`contains(null)` -- wtf here? Do not store any null objects in the list or map's keys.